x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts

```diff
@@ -76,7 +74,8 @@ const ruleTypeConfig = RULE_TYPES_CONFIG[ApmRuleType.TransactionDuration];
 export function registerTransactionDurationRuleType({
   alerting,
   ruleDataClient,
-  config$,
+  getApmIndices,
```

Comment:
Why not passing directly the indices instead of the function? Couldn't we get the indices on plugin start?

Reply:
Two reasons:

1. `registerTransactionDurationRuleType` is called when kibana starts. Retrieving the apm indicies at this point would mean that if the indices were update via APM Settings, the change would not be reflected until Kibana was restarted.

2. Because `registerTransactionDurationRuleType` is called during startup it is called with the privileges of the internal kibana user. I don't think this is a problem in practice but it feels wrong.